services/api/lib/postgre.js

```diff
@@ -295,6 +295,334 @@ module.exports = function (pg) {
         });
       });
 
+      // matches string like '$0.id' where '$0' is the action result at the 0th index and the value on that object keyed with 'id'
+      var reachRegex = /^\$(\d+)\.(.*)$/;
+
+      // Build an array of values to pass to each type of query that can be run in a bulk transaction
+      // array values are determined by type and method.
+      function getValuesForQuery(userId, type, method, data) {
+        var values;
+        switch (type) {
```

Comment:
rather than `switch(type)`, let's use an actual object
```
var handlers = {
  projects: ...,
  update: ...,
  ...
}
```
and then we don't need a switch, we simply execute `handlers[type]`. Ideally, we can remove the need for any switches that way.